### ISSUE TITLE
lowercase CN on terraform import results in broken state

### ISSUE THREAD

**Issue by Phil1602** [Ouest-France/terraform-provider-ldap#98]
Hi,

first of all: Thanks for this provider :)!

We recently run into a problem when using this provider.

The following import statement results in an analogue state. (`cn` attribute in lowercase). According to the LDAP specification this should not be an issue.

```sh
terraform import ldap_group.mygroup 'cn=mycn,OU=<redacted>,OU=<redacted>,OU=<redacted>,DC=<redacted>'
```

However when reading the resource, the LDAP provider will query for an `OU` starting with `mycn`, because it does not recognize any `CN` attribute within the delivered string.

The reason therefor is probably the recognition of the `OU` based on the following lines of code:

https://github.com/Ouest-France/terraform-provider-ldap/blob/master/ldap/resource_ldap_group.go#L143

```go
// Remove the `CN=<group-name>,` from the DN to get the OU
ou := strings.ReplaceAll(dn, fmt.Sprintf("CN=%s,", attributes["name"][0]), "")
if err := d.Set("ou", ou); err != nil {
  return diag.FromErr(err)
}
```

The `CN` is removed properly to get the `OU`,  but only if the attribute `CN` has been written uppercase to the state. This is likely the case for creation, but not for an import.

The solution is quite simple: Use uppercase syntax on import. But it would be an improvement to be more fail-safe. Solutions could be either removal of `CN` case-insensitive or maybe writing uppercase `CN` to the state on import.

**Comment by pablo-ruth** [comment]
Hi @Phil1602,

Thanks for reporting this bug, I merged a fix and released the [v0.8.9](https://github.com/Ouest-France/terraform-provider-ldap/releases/tag/v0.8.9).

**Comment by Phil1602** [comment]
Hi @pablo-ruth,

Thanks for the fast response & fix!
